### PR TITLE
Deterministic Training Support

### PR DESCRIPTION
**Pull request by klemen1999**
Added option for full reproducibility of training by specifying the seed.
This also includes deterministic order of creating `Nodes` and using CPU for drawing segmentation masks for visualizations (related to this [torchvision bug](https://github.com/pytorch/pytorch/issues/104217#issuecomment-1612967519))

### CODE REVIEW

File: luxonis_train/core/core.py [hunk 1]
```diff
@@ -90,6 +90,11 @@ def __init__(
         # NOTE: overriding logger in pl so it uses our logger to log device info
         rank_zero_module.log = logger
 
+        deterministic = False
+        if self.cfg.trainer.seed is not None:
+            pl.seed_everything(self.cfg.trainer.seed, workers=True)
+            deterministic = True
+
         self.train_augmentations = TrainAugmentations(
```
Comment (tersekmatija) [inline luxonis_train/core/core.py:98]: I understand TrainAugmentations will be deterministic due to pl.seed_everything correct?

**Review by tersekmatija — APPROVED** [review]
Looks good, thanks!

File: luxonis_train/core/core.py (hunk 1 above)
Comment (klemen1999) [reply luxonis_train/core/core.py:98]: Yes, this sets python's random seed, numpy seed and torch seed ([here](https://pytorch-lightning.readthedocs.io/en/1.7.7/_modules/pytorch_lightning/utilities/seed.html#seed_everything))
